packages/react/src/components/cookieConsent/CookieConsentContext.tsx

```diff
@@ -0,0 +1,245 @@
+  expandAriaLabel: string;
+  checkboxAriaDescription?: string;
+  consents: ConsentData[];
+};
+
+export type UiTexts = {
+  showSettings: string;
+  hideSettings: string;
+  approveAllConsents: string;
+  approveRequiredAndSelectedConsents: string;
+  approveOnlyRequiredConsents: string;
+  settingsSaved: string;
+};
+
+export type SectionTexts = {
+  main: Description;
+  details: Description;
+};
+
+export type RequiredOrOptionalConsentGroups = Description & {
+  checkboxAriaDescription?: string;
+  groupList: ConsentGroup[];
+};
+
+export type Content = {
+  requiredConsents?: RequiredOrOptionalConsentGroups;
+  optionalConsents?: RequiredOrOptionalConsentGroups;
+  texts: {
+    sections: SectionTexts;
+    ui: UiTexts;
+    tableHeadings: TableData;
+  };
+  language: {
+    languageOptions: { code: string; label: string }[];
+    current: string;
+    languageSelectorAriaLabel: string;
+    onLanguageChange: (newLanguage: string) => void;
+  };
+  onAllConsentsGiven?: (consents: ConsentObject) => void;
+  onConsentsParsed?: (consents: ConsentObject, hasUserHandledAllConsents: boolean) => void;
+};
+
+export type CookieConsentContextType = {
+  update: ConsentController['update'];
+  approveRequired: ConsentController['approveRequired'];
+  approveAll: ConsentController['approveAll'];
+  hasUserHandledAllConsents: () => boolean;
+  content: Content;
+  onAction: CookieConsentActionListener;
+  countApprovedOptional: () => number;
+  areGroupConsentsApproved: (consents: ConsentData[]) => boolean;
+};
+
+type CookieConsentContextProps = {
+  cookieDomain?: string;
+  children: React.ReactNode | React.ReactNode[] | null;
+  content: Content;
+};
+
+export const CookieConsentContext = createContext<CookieConsentContextType>({
+  update: () => undefined,
+  approveRequired: () => undefined,
+  approveAll: () => undefined,
+  hasUserHandledAllConsents: () => false,
+  content: {} as Content,
+  onAction: () => undefined,
+  countApprovedOptional: () => 0,
+  areGroupConsentsApproved: () => false,
+});
+
+export const getConsentsFromConsentGroup = (groups: ConsentGroup[]): ConsentList => {
+  return groups.reduce((ids, currentGroup) => {
+    currentGroup.consents.forEach((consentData) => {
+      ids.push(consentData.id);
+    });
+    return ids;
+  }, []);
+};
+
+export const Provider = ({ cookieDomain, children, content }: CookieConsentContextProps): React.ReactElement => {
+  const requiredConsents = content.requiredConsents
+    ? getConsentsFromConsentGroup(content.requiredConsents.groupList)
+    : undefined;
+  const optionalConsents = content.optionalConsents
+    ? getConsentsFromConsentGroup(content.optionalConsents.groupList)
+    : undefined;
+  const consentController = useMemo(() => create({ requiredConsents, optionalConsents, cookieDomain }), []);
+  const hasUserHandledAllConsents = () =>
+    consentController.getRequiredWithoutConsent().length === 0 && consentController.getUnhandledConsents().length === 0;
```

Comment:
Should we combine the functions `getRequiredWithoutConsent` and `getUnhandledConsents `? They seem to only be used in this line, and reading the code here is little misleading. As we have a function named `getUnhandledConsents`, one could expect that calling it would suffice to find out whether user has handled all consents or not? Maybe we combine them under the name `getUnhandledConsents`, what do you think?